src/main/java/org/jenkinsci/plugins/docker/traceability/util/FingerprintsHelper.java

```diff
@@ -62,13 +62,21 @@ public static void addFacet(@Nonnull Fingerprint fingerprint, @Nonnull Fingerpri
     
     /**
      * Get all imageIDs, which have fingerprints somewhere. 
+     * This method iterates through {@link Item}s and {@link Run}s in order to
+     * find actions with image fingerprint references, there is no any caching of the data.
      * @return Collection of image IDs, which have declared fingerprints.
+     * @deprecated Not recommended for a use due to the performance issues.
+     *      A better method will be implemented in docker-commons at some point
      */
-    public static Set<String> getImagesWithFingerprints() {
-        Set<String> result = new HashSet<String>();
+    @Deprecated
+    public static @Nonnull Set<String> getImagesWithFingerprints() {
+        final Set<String> result = new HashSet<String>();
+        final Jenkins j = Jenkins.getInstance();
+        if (j == null) {
+            return result;
+        }
 
-        //TODO: Just4test, horrible performance, we need caching
```

Comment:
I've marked the method as Deprecated with details and also got rid of it's usages within  the plugin, so there's no TODO now.

BTW, it could make sense just to delete this method at all